### PR TITLE
[Backend] - Update Cart API

### PR DESCRIPTION
**Pull request by TreshAnn**
## Ticket/s

- [Update Cart API](https://trello.com/c/hnKHmcH3)

## Changes

- Add Update Cart Item API

## Screenshot
- Current Cart Document 
image

- Updated Product Quantity 
image
image

## How to test?

- Open Postman
- click import tab, paste cURL
```
curl --location --request PUT 'http://localhost:3001/api/cart' \
--header 'Authorization: Bearer [insert access token here]' \
--header 'Content-Type: application/json' \
--data '{
    "productID": "[insert productID here]",
    "quantity": 3
}'
```
Note: Must have user data and sample product before testing
- Log in as a user to obtain the access token
- Copy the obtained token to the Auhorization Header
- Update the `productID` and `quantity` as needed

### CODE REVIEW

File: apps/api/src/cart/cart.service.ts [hunk 1]
```diff
@@ -35,6 +37,10 @@ export class CartService {
 
   async createItem(productID: string, quantity: number) {
     const product = await this.productsService.findOne(productID);
+
+    if (!product) {
+      throw new NotFoundException('Product not found.');
+    }
```
Comment (brentmillen) [inline apps/api/src/cart/cart.service.ts:43]: The exception is already in the productService function. Can you test if it still works correctly without this condition?

File: apps/api/src/cart/cart.service.ts [hunk 2]
```diff
@@ -99,6 +105,32 @@ export class CartService {
...
     }
   }
+  async updateCartItem(userID: string, itemDto: ItemDto): Promise<Cart> {
+    const { productID, quantity, productPrice } = itemDto;
+
+    const cart = await this.getCart(userID);
+
+    if (quantity <= 0) {
+      throw new BadRequestException('Quantity must be greater than 0.');
+    }
```
Comment (brentmillen) [inline apps/api/src/cart/cart.service.ts]: just reuse the validateQuantity for this one

**Review by sandrex-SGV — COMMENTED** [review]
Please update PR's testing instructions.

**Comment by TreshAnn** [comment]
> Please update PR's testing instructions.

Done. Thanks, sir drex!

**Review by brentmillen — CHANGES_REQUESTED** [review]

File: apps/api/src/cart/cart.service.ts (hunk 1 above)
Comment (TreshAnn) [reply apps/api/src/cart/cart.service.ts:43]: image

File: apps/api/src/cart/cart.service.ts (hunk 2 above)
Comment (TreshAnn) [reply apps/api/src/cart/cart.service.ts]: Done, Sir! Thanks
image

**Review by sandrex-SGV — APPROVED** [review]
yields expected results.
LGTM!

**Review by brentmillen — APPROVED** [review]

**Pull request merged**